core-java/src/main/java/eu/ostrzyciel/jelly/core/JellyException.java

```diff
@@ -0,0 +1,41 @@
+package eu.ostrzyciel.jelly.core;
+
+public sealed class JellyException extends RuntimeException {
+
+    public static JellyException rdfProtoDeserializationError(String msg) {
+        return new RdfProtoDeserializationError(msg);
+    }
+
+    public static JellyException rdfProtoSerializationError(String msg) {
+        return new RdfProtoSerializationError(msg);
+    }
+
+    public static JellyException rdfProtoTranscodingError(String msg) {
+        return new RdfProtoTranscodingError(msg);
+    }
+
+    public JellyException(String message) {
+        super(message);
+    }
+
+    public static final class RdfProtoDeserializationError extends JellyException {
```

Comment:
Maybe move these classes to top level and remove the factory methods.

The factories were there just for Java compat.

Reply:
move to top level

Reply:
doned